nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/SessionImpl.java

```diff
@@ -1241,14 +1253,17 @@ public Node getLastVersion(Serializable versionSeriesId) {
                 nodes.add(node);
             }
         }
+        return nodes;
+    }
 
-        if (parent != null) {
-            // filter by parent
-            Serializable parentId = parent.getId();
-            nodes.removeIf(node -> !parentId.equals(node.getParentId()));
+    @Override
+    public List<Node> getProxies(Node document) {
+        checkLive();
+        if (!repository.getRepositoryDescriptor().getProxiesEnabled()) {
+            return Collections.emptyList();
         }
-
-        return nodes;
+        List<Serializable> ids = context.getSeriesProxyIds(document.getId());
```

Comment:
You want `getTargetProxyIds` here, not `getSeriesProxyIds` which return proxies for the full series.
Maybe in the test add an extra unrelated proxy+version in the series to make sure you don't get it when requesting the proxies for a given version/doc.